lib/nerves_hub/devices.ex

```diff
@@ -698,20 +698,22 @@ defmodule NervesHub.Devices do
 
   @spec update_deployment(Device.t(), Deployment.t()) :: Device.t()
   def update_deployment(device, deployment) do
-    device
-    |> Ecto.Changeset.change()
-    |> Ecto.Changeset.put_change(:deployment_id, deployment.id)
-    |> Repo.update!()
-    |> Repo.preload(:deployment, force: true)
+    %Device{} =
+      device
+      |> Ecto.Changeset.change()
+      |> Ecto.Changeset.put_change(:deployment_id, deployment.id)
+      |> Repo.update!()
+      |> Repo.preload(:deployment, force: true)
```

Comment:
Im guessing this is a unique work around to `Repo.preload` allowing returns of list, struct, or nil. But I would vote we learn from dialyzer to rethink the approach here rather than add something that passes but is visually odd and maybe confusing to others.

i.e. it seems what we're really trying to accomplish is ensuring the deployment is in the struct. So why not just put it in there rather than requiring a DB reload?
```suggestion
    %Device{} =
      device
      |> Ecto.Changeset.change()
      |> Ecto.Changeset.put_change(:deployment_id, deployment.id)
      |> Repo.update!()
      |> Map.put(:deployment, deployment)
```

Reply:
Fair call, I'll merge this in and check it out.